xbmc/platform/win32/filesystem/Win32WSDiscovery.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ *  Copyright (C) 2005-2021 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "Win32WSDiscovery.h"
+
+
+HRESULT CreateClientNotificationSink(CClientNotificationSink** sink)
+{
+  CClientNotificationSink* tempSink = nullptr;
+
+  if (!sink)
+    return E_POINTER;
+
+  tempSink = new CClientNotificationSink();
+
+  if (!tempSink)
+    return E_OUTOFMEMORY;
+
+  *sink = tempSink;
+  tempSink = nullptr;
+
+  return S_OK;
+}
+
+CClientNotificationSink::CClientNotificationSink() : m_cRef(1)
+{
+  m_finished = false;
+  m_serversIps.clear();
+}
```

Comment:
iirc we prefer default member initializers and is clearing the vector needed?

Reply:
True, it is not needed. In the new update is removed also `m_finished` is not used now.

Reply:
cool, thanks. why not `ULONG m_cRef = 1` in line 43 and remove ctor and dtor? just asking

Reply:
I really could not say why but it is a habit that I have to do so. I understand that it is a good codding practice to declare ctor and dtor even if they are empty although I am not an expert in c++ in things at a theoretical level.